install/kubernetes/helm/subcharts/gateways/templates/deployment.yaml

```diff
@@ -143,7 +143,11 @@ spec:
           - name: ISTIO_META_POD_NAME
             valueFrom:
               fieldRef:
-                fieldPath: metadata.name             
+                fieldPath: metadata.name  
+          {{- if $.Values.global.network }}
+          - name: ISTIO_META_NETWORK
+            value: "{{ $.Values.global.network }}"
+          {{- end }}           
```

Comment:
Once again no :). Please do not encode anything to do with a network into the gateways. You will force every user of custom gateways to use this, which is not a good experience as it adds to the bloat. Take a look at the values-istio-multicluster-gateways.yaml to see how we add the environment variable under the values section.